### PR TITLE
[PAY-1979] Update Backing Mutation To Use Setup Intent

### PR DESCRIPTION
**Pull request by msadoon**
# 📲 What

In order to support the pre-existing use cases of Update Backing Mutation on iOS pledge page
- `update`
- `changePaymentMethod`
- `updateReward`
we need to allow the setup intent client secret to be passed to `UpdateBackingType` mutation.

# 🤔 Why

With the `.pledge` case covered in our existing implementation of the Stripe payment sheet, we continue that support for the other cases where a live pledge is modified.

# 🛠 How

Not much additional work except to include `.update`, `.updateReward`, `.changePaymentMethod` alongside `.pledge` inside the `paymentSheetOnPledgeContext` signal in `PledgePaymentMethodsViewModel`.

# 👀 See

Before 🐛 

https://user-images.githubusercontent.com/4282741/194437280-78406887-03d8-406f-a8fb-1bfa2b4835f4.mp4

After 🦋

https://user-images.githubusercontent.com/4282741/194437267-ec894a0d-3d70-4b46-aeb7-1716faeab058.mp4

# ✅ Acceptance criteria

- [x] Ensure all scenarios except (`.fixPaymentMethod`) are working when FF is ON (`.update`, `.changePaymentMethod`, `.updateReward)
- [x] Ensure all prexisting flows are not changed when FF is OFF (`.update`, `.changePaymentMethod`, `.updateReward`, `.fixPaymentMethod`) 

# ⏰ TODO

- [x] Update mutation for accepting setup intent
- [x] Add business logic for supporting new mutation parameter
- [x] Write tests.

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: KsApi/GraphAPI.swift [hunk 1]
```diff
@@ -723,57 +724,6 @@ public enum GraphAPI {
     }
   }
 
-  /// Necessary fields for Google Pay
```
Comment (scottkicks) [inline KsApi/GraphAPI.swift:726]: Why aren't we supporting Google Play anymore?

Comment (msadoon) [reply KsApi/GraphAPI.swift:726]: Good question, not 100% sure why, I don't think it was ever fully supported. Also _I think_ there is potential for the Stripe payment sheet to handle both Apple Pay and Google Pay and many more payment options. Just my hunch, Isa is probably more knowledgeable about that.

Comment (joshua-ksr) [reply KsApi/GraphAPI.swift:726]: Some quick historical context that we previously supported Android Pay in a limited capacity, but don't think we actually ever supported Google Pay. +1 to Mubarak that Google Pay will be a good opportunity to us post-Payment Sheet implementation.

**Review by scottkicks — APPROVED** [review]

File: KsApi/GraphAPI.swift (hunk 1 above)
Comment (scottkicks) [reply KsApi/GraphAPI.swift:726]: Got it. Thanks for providing more context around this

**Pull request merged**